### PR TITLE
Adding a rocket simulator.

### PR DESCRIPTION
**Pull request by teytaud**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](https://facebookresearch.github.io/nevergrad/contributing.html) document and completed the CLA (see [**CLA**](https://facebookresearch.github.io/nevergrad/contributing.html#contributor-license-agreement-cla)).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: nevergrad/functions/rocket/rocket.py [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+        time += dt
+        update(Ex, Ey, Ez, Evx, Evy, Evz, time, r)
+    
+    # After thrust
+    # This is when the engine is out of fuel and there is no longer a thrust force
+    time_step = .05  # time time_step in seconds
+    dt = time_step
+    while r > r_initial:
+        r = (Ex ** 2 + Ey ** 2 + Ez ** 2) ** 0.5
+        Efx, Efy, Efz = net_force(Ex, Ey, Ez, Evx, Evy, Evz, final_roc_mass)
```
Comment (jrapin) [inline nevergrad/functions/rocket/rocket.py:274]: this all looks like it should be numpy arrays to:
- require less lines of code
- be much faster

Comment (teytaud) [reply nevergrad/functions/rocket/rocket.py:271]: Replaced lists by numpy arrays.
Did not optimize anything for now.

File: nevergrad/functions/rocket/rocket.py [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+
+"""
+Approximate Rocket Simulation
+Based on 
+https://raw.githubusercontent.com/purdue-orbital/rocket-simulation/master/Simulation2.py
+"""
+
+
+import math
+import pyproj  # type: ignore
```
[the remark this answers is not in the document]
Comment (teytaud) [reply nevergrad/functions/rocket/rocket.py]: done.

File: nevergrad/functions/rocket/rocket.py [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+def rocket(thrust_bias: np.ndarray):
+
+    assert len(thrust_bias) == 24, "Bad guide length."
+    # Covert ang to rads
+    def rad(ang):
+        return (ang / 360) * 2 * (3.1415926)
+    
+    
+    # air density (simple model based on nasa function online)
+    def air_density(alt):
```
[the remark this answers is not in the document]
Comment (teytaud) [reply nevergrad/functions/rocket/rocket.py:27]: I have a dilemma here.. should I get farther and farther from the original code ?

[the remark this answers is not in the document]
Comment (jrapin) [reply nevergrad/functions/rocket/rocket.py:27]: ah i did not realize it was taken from somewhere, maybe no need then

**Comment by teytaud** [comment]
The bug in BO has nothing to do with the current PR, maybe...

**Comment by jrapin** [comment]
> The bug in BO has nothing to do with the current PR, maybe...

There is no BO bug here, this is naiveseqmltuning taking way too long for the CI

**Comment by teytaud** [comment]
> > The bug in BO has nothing to do with the current PR, maybe...
> 
> There is no BO bug here, this is naiveseqmltuning taking way too long for the CI

Maybe you're right (usually when we disagree at the end it turns out I was wrong) but it really looooks like a BO issue:

================================== FAILURES ===================================
_____________________ test_optimizers_recommendation[BO] ______________________

self = Instance of BO(parametrization=Array{(4,)}, budget=6, num_workers=1)

    def _internal_ask_candidate(self) -> p.Parameter:
        util = UtilityFunction(kind=self.utility_kind, kappa=self.utility_kappa, xi=self.utility_xi)
        try:
>           x_probe = next(self.bo._queue)

nevergrad\optimization\optimizerlib.py:1294: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <bayes_opt.bayesian_optimization.Queue object at 0x000001CEFCF77188>

    def __next__(self):
        if self.empty:
>           raise StopIteration("Queue is empty, no more objects to retrieve.")
E           StopIteration: Queue is empty, no more objects to retrieve.

venv\lib\site-packages\bayes_opt\bayesian_optimization.py:25: StopIteration

During handling of the above exception, another exception occurred:

name = 'BO'
recomkeeper = <nevergrad.optimization.test_optimizerlib.RecommendationKeeper object at 0x000001CE801B5F48>

    @pytest.mark.parametrize("name", registry)  # type: ignore
    def test_optimizers_recommendation(name: str, recomkeeper: RecommendationKeeper) -> None:
        # set up environment
        optimizer_cls = registry[name]
        if name in UNSEEDABLE:
            raise SkipTest("Not playing nicely with the tests (unseedable)")
        np.random.seed(None)
        if optimizer_cls.recast or "SplitOptimizer" in name:
            np.random.seed(12)
            random.seed(12)  # may depend on non numpy generator
        # budget=6 by default, larger for special cases needing more
        budget = {"WidePSO": 100, "PSO": 200, "MEDA": 100, "EDA": 100, "MPCEDA": 100, "TBPSA": 100}.get(name, 6)
        if isinstance(optimizer_cls, (optlib.DifferentialEvolution, optlib.EvolutionStrategy)):
            budget = 80
        dimension = min(16, max(4, int(np.sqrt(budget))))
        # set up problem
        fitness = Fitness([0.5, -0.8, 0, 4] + (5 * np.cos(np.arange(dimension - 4))).tolist())
        with warnings.catch_warnings():
            # tests do not need to be efficient
            warnings.filterwarnings("ignore", category=base.InefficientSettingsWarning)
            optim = optimizer_cls(parametrization=dimension, budget=budget, num_workers=1)
            optim.parametrization.random_state.seed(12)
            np.testing.assert_equal(optim.name, name)
            # the following context manager speeds up BO tests
            # BEWARE: BO tests are deterministic but can get different results from a computer to another.
            # Reducing the precision could help in this regard.
            patched = partial(acq_max, n_warmup=10000, n_iter=2)
            with patch("bayes_opt.bayesian_optimization.acq_max", patched):
>               recom = optim.minimize(fitness)

**Pull request merged**